internal/pkg/flink/internal/controller/statement_controller.go

```diff
@@ -125,14 +113,33 @@ func (c *StatementController) waitForStatementToBeInTerminalStateOrError(process
 	return terminalStatement, nil
 }
 
-func (c *StatementController) isDetachEvent() bool {
-	if b, err := c.consoleParser.Read(); err == nil && len(b) > 0 {
-		pressedKey := prompt.Key(b[0])
+func (c *StatementController) userInputIsOneOf(keyEvents ...func(key prompt.Key) bool) func() bool {
+	return func() bool {
+		if b, err := c.consoleParser.Read(); err == nil && len(b) > 0 {
+			pressedKey := prompt.Key(b[0])
 
-		switch pressedKey {
-		case prompt.ControlM, prompt.Enter:
-			return true
+			for _, isKeyEvent := range keyEvents {
+				if isKeyEvent(pressedKey) {
+					return true
+				}
+			}
 		}
+		return false
+	}
+}
+
+func isCancelEvent(key prompt.Key) bool {
+	switch key {
+	case prompt.ControlC, prompt.ControlD, prompt.ControlQ, prompt.Escape:
+		return true
+	}
+	return false
+}
+
+func isDetachEvent(key prompt.Key) bool {
+	switch key {
```

Comment:
and then you could also reduce this to one line:
`lo.Contains([]prompt.Key{prompt.ControlM, prompt.Enter}, key)`